app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -90,6 +92,26 @@ class HomeFragment : Fragment() {
         }
 
         registerForContextMenu(view)
+
+        tileContainer.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                // Only add scroll offset when user is scrolling down
+                if (dy > 0) {
+                    val layoutManager = recyclerView?.layoutManager as GridLayoutManager
```

Comment:
You don't have to use GridLayoutManager, can use a parent because you don't use any GridLayoutManager-specific calls.

Reply:
What do u mean by use a parent, I used GridLayoutManager because I needed access to those APIs for the children stuff

Reply:
Here's the docs for GridLayoutManager - at the top is its inheritance chain. If you use GridLayoutManager, that's more constrained than it needs to be, its parents have have methods for `childCount`, etc as well. Unless there's one of these calls that is Grid-specific? In general, try to use the most general parent as possible.

https://developer.android.com/reference/android/support/v7/widget/GridLayoutManager.html